language/vm/vm-runtime/src/system_txn/block_metadata_processor.rs

```diff
@@ -46,16 +45,21 @@ where
     let gas_schedule = CostTable::zero();
 
     // The transaction executor failed to start
-    let mut txn_executor =
-        TransactionExecutor::new(&module_cache, &gas_schedule, data_cache, txn_data);
+    let mut txn_executor = TransactionExecutor::new(&gas_schedule, data_cache, txn_data);
     let result = if let Ok((id, timestamp, previous_vote, proposer)) = block_metadata.into_inner() {
         let args = vec![
             Value::u64(timestamp),
             Value::byte_array(id),
             Value::byte_array(previous_vote),
             Value::address(proposer),
         ];
-        txn_executor.execute_function(&LIBRA_SYSTEM_MODULE, &BLOCK_PROLOGUE, args)
+        txn_executor.execute_function(
```

Comment:
Can we use `runtime` to execute the function rather than using `TransactionExecutor`?

Reply:
let me look into that and I see where you are coming from and good point!

Reply:
I looked at this because I think it's a really good idea. However it makes changes to the way the write set is computed (it lives in `TransactionExecutor` now) so I am leaving it to you.
I am not sure how you are going to reorganize that code and you are the best one to make that call